src/audio/volume/volume.c

```diff
@@ -408,57 +423,56 @@ static int volume_ctrl_set_cmd(struct comp_dev *dev,
 				      "cdata->comp_id = %u",
 				      cdata->comp_id);
 		for (j = 0; j < cdata->num_elems; j++) {
-			trace_volume_with_ids(dev, "volume_ctrl_set_cmd(), "
-					      "SOF_CTRL_CMD_VOLUME, "
-					      "channel = %u, value = %u",
-					      cdata->chanv[j].channel,
-					      cdata->chanv[j].value);
-			i = cdata->chanv[j].channel;
-			if (i >= 0 && i < SOF_IPC_MAX_CHANNELS) {
-				ret = volume_set_chan(dev, i,
-						      cdata->chanv[j].value);
-			} else {
+			ch = cdata->chanv[j].channel;
+			val = cdata->chanv[j].value;
+			trace_volume_with_ids(dev, "volume_ctrl_set_cmd(), channel = %d"
+					      ", value = %u", ch, val);
+			if (ch < 0 || ch >= SOF_IPC_MAX_CHANNELS) {
 				trace_volume_error_with_ids(dev,
-							    "volume_ctrl_set_cmd() "
-							    "error: "
-							    "SOF_CTRL_CMD_VOLUME, "
-							    "invalid i = %u",
-							    i);
+							    "volume_ctrl_set_cmd(), illegal channel = %d",
+							    ch);
+				return -EINVAL;
+			}
+
+			if (cd->muted[ch]) {
+				cd->mvolume[ch] = val;
+			} else {
+				ret = volume_set_chan(dev, ch, val);
```

Comment:
You could also move the declaration here
```
int ret = volume_set_...();
```
to make it clear that `ret` isn't needed anywhere else. But probably not worth redoing the patch just for this.

Reply:
Unless I'm mistaken that's not done in SOF. I'll fix the typo any way, it's safe to change without need for me to re-test stuff.

Reply:
IMHO what @lyakh suggested is better, BUT @singalsu is right that it's forbidden in SOF. In kernel you can see that they allow declaration on start of the block, but in sound tree there is rule that everything goes to the start of the function, because most of maintainers decided so.